scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+In theory, I think, it's pretty easy to do in Scala anything that's easy to do in Java. Some of that may not be "good" Scala, though. Basic object-oriented programming, for example isn't any more difficult (even if the idea of companion objects is kind of weird at first). Functional programming is pretty easy in Scala as well, although I was surprised to find that higher-order functions are typically methods of types for which those functions make any sense. I found that it was
+quite difficult to use recursion and pattern-matching to do something better handled without the pattern-matching. In particular, I basically gave up figuring out how to implement zipMultiple with pattern-matching, looked at the sample solution, realized _why_ I hadn't figured it out (it was considerably more complicated than I think is necessary), and ended up just referring to Prof. Ben's solution. Obviously there are going to be some tasks that pattern matching won't be suited for regardless of the
+language, but I'm not sure whether this is one of them, or if Scala's pattern matching is particularly not suited to it.
+
```

Comment:
I'm curious, just from a subjective standpoint, what your opinion is of Scala. I've heard a lot of mixed opinions, with good arguments for it being both convenient and somewhat obnoxious, so I'd like to hear more about that. Its an interesting mix of different language archetypes.